### PR TITLE
Serhii buniak/2020 12 27 parse cache control header

### PR DESCRIPTION
**Pull request by MasterSergius**

### CODE REVIEW

File: src/acachecontrol/cache.py [hunk 1]
```diff
@@ -18,18 +19,33 @@ def __init__(self):
     def __contains__(self, key):
         return self._make_key_hashable(key) in self.cache
 
-    def add(self, key: Tuple[str, str, Dict], value: Any) -> None:
+    def add(self, key: Tuple[str, str, Dict], value: Any, headers: Any):
```
Comment (andrii-ivaniuk) [inline src/acachecontrol/cache.py]:
```suggestion
    def add(self, key: Tuple[str, str, Dict], value: Any, headers: Any) -> None:
```

File: src/acachecontrol/cache.py [hunk 2]
```diff
@@ -18,18 +19,33 @@ def __init__(self):
...
         """Add value to the cache.
 
+        headers - any dict-like obj
+
         # TODO: Values should be stored within the following structure:
         {'created_at': <unix_timestamp>, 'max_age': <seconds>, 'value': <value>}
```
Comment (andrii-ivaniuk) [inline src/acachecontrol/cache.py]:
```suggestion
```

File: src/acachecontrol/cache.py [hunk 3]
```diff
@@ -18,18 +19,33 @@ def __init__(self):
...
+            "value": value,
+        }
         self.release_new_key(key)
 
     def get(self, key: Tuple[str, str, Dict]) -> Any:
-        return self.cache.get(self._make_key_hashable(key))
+        try:
+            cache_entry = self.cache.get(self._make_key_hashable(key))
+            if cache_entry:
+                return cache_entry.get("value")
```
Comment (andrii-ivaniuk) [inline src/acachecontrol/cache.py]:
```suggestion
                return cache_entry["value"]
```
Is `value` a required key?

File: src/acachecontrol/cache.py [hunk 4]
```diff
@@ -81,3 +97,31 @@ def _dict_hash(input_dict: Dict) -> str:
...
+
+    @staticmethod
+    def parse_cache_control_header(headers) -> Dict:
+        """Parse cache-control header, get max-age value.
+
+        Args:
+            headers: any dict-like object
+
+        Returns:
+            dict
```
Comment (andrii-ivaniuk) [inline src/acachecontrol/cache.py]: I think it's better to add a description of the returned value and an example.

File: tests/integration/test_acachecontrol.py [hunk 5]
```diff
@@ -10,12 +10,12 @@ def __init__(self):
...
-    def add(self, key, value):
-        super().add(key, value)
-        self.add_calls.append((key, value))
+    def add(self, key, value, headers):
+        super().add(key, value, headers)
+        self.add_calls.append((key, value, headers))
 
     def get(self, key):
-        value = self.cache.get(self._make_key_hashable(key))
+        value = self.cache.get(self._make_key_hashable(key)).get("value")
```
Comment (andrii-ivaniuk) [inline tests/integration/test_acachecontrol.py]:
```suggestion
        value = self.cache.get(self._make_key_hashable(key))["value"]
```

File: src/acachecontrol/cache.py (hunk 4 above)
Comment (MasterSergius) [reply src/acachecontrol/cache.py]: added

**Pull request merged**